packages/salesforcedx-vscode-core/src/sfdxProject/sfdxProjectConfig.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2019, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { SfdxProject, SfdxProjectJson } from '@salesforce/core';
+import { AnyJson, JsonMap } from '@salesforce/ts-types';
+import * as path from 'path';
+import * as vscode from 'vscode';
+
+import { SFDX_PROJECT_FILE } from '../constants';
+import { nls } from '../messages';
+import { notificationService } from '../notifications';
+import { isSfdxProjectOpened } from '../predicates';
+import { SfdxProjectPath } from '../sfdxProject';
+import { telemetryService } from '../telemetry';
+
+/**
+ * Class representing the local sfdx-project.json file.
+ * Does not contain global values.
+ */
+export default class SfdxProjectConfig {
+  private static instance: SfdxProjectJson;
+  private constructor() {
+    throw new Error(
+      'Error: *** call SfdxProject.getInstance() to get the singleton instance of this class ***'
+    );
+  }
+
+  private static async initializeSfdxProjectConfig() {
+    if (
+      !SfdxProjectConfig.instance &&
+      isSfdxProjectOpened.apply(vscode.workspace).result
+    ) {
+      const sfdxProjectPath = SfdxProjectPath.getPath();
+      try {
+        const sfdxProject = await SfdxProject.resolve(sfdxProjectPath);
+        SfdxProjectConfig.instance = await sfdxProject.retrieveSfdxProjectJson();
```

Comment:
Previously, we were using `sfdxProject.resolveProjectConfig()`, which fetches an aggregation of the global + local config values. Using `sfdxProject.retrieveSfdxProjectJson()` instead will only pull in the values present in the local `sfdx-project.json` config file.

Reply:
With this new way do you think default values for missing entries (sfdcLoginUrl for example) will still be returned? Was that case actually a default value, or something from a global config?

Reply:
@brpowell If a value isn't found in the sfdx-project.json, then getValue() will just return `undefined`. I tested the auth command, and when the sfdx-project.json doesn't contain a `sfdcLoginUrl`, then no default is provided in the dropdown.